apps/status-service/src/app/jobs/checkEndpoint.ts

```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
 
   return response;
 };
+
+/**
+ * This is an attached job that checks auto change status boolean flag and determines
+ * what status to update based on the health check
+ */
+async function checkAndUpdateAutoChangeStatus(props: CreateCheckEndpointProps) {
+  const { app, serviceStatusRepository, endpointStatusEntryRepository, logger } = props;
+  const QUERY_SIZE = 5;
+
+  try {
+    if (app.autoChangeStatus) {
+      //The history array contains 1 item for each minute.
+      const recentHistory = await endpointStatusEntryRepository.findRecentByUrlAndApplicationId(
+        app.url,
+        app.appKey,
+        QUERY_SIZE
+      );
+
+      //get first and last date time to make sure the time is 5 minutes apart.
+      const mostRecentDate = moment(new Date(recentHistory.at(0).timestamp));
+      const oldestDate = moment(new Date(recentHistory.at(-1).timestamp));
+
+      //Need to round date diff as MomentJS can give you an incorrect number because of the milliseconds
+      const dateDiff = Math.round(mostRecentDate.diff(oldestDate, 'minute'));
+
+      const failedForMoreThanFiveMinutes = recentHistory.filter((hist) => !hist.ok && dateDiff >= 5).length > 0;
+      const succeededMoreThanFiveMinutes = recentHistory.filter((hist) => hist.ok && dateDiff >= 5).length > 0;
+      const currentServiceStatus = await serviceStatusRepository.get(app.appKey);
+
+      //Set the status to opperational if it is in report issues status and
+      //the the health check has failed for the last five minutes
+      if (
+        currentServiceStatus &&
+        currentServiceStatus.status !== '' &&
+        currentServiceStatus.status === 'operational' &&
+        failedForMoreThanFiveMinutes
+      ) {
+        //update status to reported issues
+        currentServiceStatus.status = 'reported-issues';
+        serviceStatusRepository.save(currentServiceStatus);
+      } else if (
+        currentServiceStatus &&
+        currentServiceStatus.status !== '' &&
+        currentServiceStatus.status === 'reported-issues' &&
+        succeededMoreThanFiveMinutes
+      ) {
+        //update status to operational
+        currentServiceStatus.status = 'operational';
+        serviceStatusRepository.save(currentServiceStatus);
+      }
+    }
```

Comment:
Suggestion: refactor for readability.  Something like:
if (currentServiceStatus && currentServiceStatus.status !-- '') {
  checkStatusForApplicationFailure(...)
  checkStatusForApplicationRevival(...)
}

Reply:
Err... Pauls suggestion for testing the status is correct.  You only need to check currentServiceStatus?.status.  I can never remember what language I'm in :(